src/platform/i386/boot_comp.c

```diff
@@ -12,6 +12,9 @@
 
 extern u8_t *boot_comp_pgd;
 
+void *thd_mem[NUM_CPU], *tcap_mem[NUM_CPU];
```

Comment:
Looks to me like we don't need these anymore.  They can just be local in `kern_boot_comp`.  Can you add an issue to fix this?